Sage/carbon
# refactor(portal): replace scss with styled-component

### Proposed behaviour
`Portal` component - Remove SCSS. The z-index isn't used and the positioning style are only used in `Toast` and tooltips, so I've added a `StyledPortal` with them on in the tooltip decorator.

Swap z-index around of full screen modal and modals in theme config (the modal one wasn't even used until now, where I have added it into `Dialog`) - this is to keep the nested dialog functionality working as expected, and how it should have worked in the first place.

Also replace `package.json` with `index.js`.

### Current behaviour
N/A

### Checklist


### Additional context

### Testing instructions
Look at component stories which use portal, for example `Alert`, `Confirm`, `Dialog` etc
